contrib/buildozer/src/python/pants/contrib/buildozer/buildozer.py

```diff
@@ -0,0 +1,68 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.binaries.binary_util import BinaryUtil
+from pants.task.task import Task
+from pants.util.process_handler import subprocess
+
+
+class Buildozer(Task):
+  """Enables interaction with the Buildozer Go binary
+
+  Behavior:
+  1. `./pants buildozer --add-dependencies=<dependencies> --location=<directory>`
+      will add the dependency to the location's relative BUILD file.
+
+      Example: `./pants buildozer --add-dependencies='a/b b/c' //tmp:tmp`
+
+  2. `./pants buildozer --remove-dependencies=<dependencies> --location=<directory>`
+      will remove the dependency from the location's relative BUILD file.
+
+      Example: `./pants buildozer --remove-dependencies='a/b b/c' //tmp:tmp`
+
+  Note that buildozer assumes that BUILD files contain a name field for the target.
+  """
+
+  @classmethod
+  def register_options(cls, register):
+    register('--version', default='0.4.5', help='Version of buildozer.')
+    register('--add-dependencies', type=str, help='The dependency or dependencies to add')
+    register('--remove-dependencies', type=str, help='The dependency or dependencies to remove')
+
+  def __init__(self, *args, **kwargs):
+    super(Buildozer, self).__init__(*args, **kwargs)
+
+    self.options = self.get_options()
+    self.address = self.context.target_roots[0].address
```

Comment:
1. We want to run buildozer against all specified target roots, not just the first one. buildozer allows your to append targets as well. 
```
buildozer 'add dependencies 456' examples/tests/java/org/pantsbuild/example/hello/greet:greet examples/tests/java/org/pantsbuild/example/useproto:proto
```
2. `self.address` as a instance variable seems unnecessary because it is only used by one other function. so it would be more clear to pass it as an argument to `_execute_buildozer_script` or just have `_execute_buildozer_script` to do the deeds.